backend/test/til_web/controllers/post_controller_test.exs

```diff
@@ -0,0 +1,311 @@
+defmodule TilWeb.PostControllerTest do
+  use TilWeb.ConnCase
+  import Til.Guardian
+  import Til.Factory
+  alias Til.ShareableContent
+
+  describe "GET /api/posts" do
+    test "returns all existing posts with categories as public", %{conn: conn} do
+      first_category = insert(:category, name: "Elixir")
+      second_category = insert(:category, name: "Javascript")
+
+      insert(:post, categories: [first_category, second_category])
+      insert(:post, categories: [first_category])
+
+      response =
+        conn
+        |> get(Routes.post_path(conn, :index))
+
+
+      assert response.status == 200
+
+      {:ok, parsed_response_body} = Jason.decode(response.resp_body)
+
+      [first_post, second_post] = parsed_response_body
+
+      assert first_post["categoriesIds"] == [first_category.id, second_category.id]
+      assert second_post["categoriesIds"] == [first_category.id]
+    end
+  end
+
+  describe "GET /api/posts/:id" do
+    test "returns particular post with categories as public", %{conn: conn} do
+      first_category = insert(:category, name: "Elixir")
+      second_category = insert(:category, name: "Javascript")
+      post_title = "Some post"
+
+      post = insert(:post, title: post_title, categories: [first_category, second_category])
+
+      response =
+        conn
+        |> get(Routes.post_path(conn, :show, post.id))
+
+      assert response.status == 200
+
+      {:ok, parsed_response_body} = Jason.decode(response.resp_body)
+
+      assert parsed_response_body["title"] == post_title
+      assert parsed_response_body["categoriesIds"] == [first_category.id, second_category.id]
+    end
+  end
+
+  describe "POST /api/posts" do
+    test "creates post and returns 201 status with created post", %{conn: conn} do
+      current_user = insert(:user)
+      {:ok, token, _} = encode_and_sign(current_user.uuid, %{})
+
+      post_title = "Some post title"
+      post_body = "Some post body"
+
+      response =
+        conn
+        |> put_req_header("authorization", "bearer: " <> token)
+        |> post(Routes.post_path(conn, :create), %{
+          title: post_title,
+          body: post_body,
+          categoriesIds: []
+        })
+
+      assert response.status == 201
+
+      {:ok, parsed_response_body} = Jason.decode(response.resp_body)
+
+      created_post = ShareableContent.get_post_by(title: post_title)
```

Comment:
not sure if we should use our module to fetch data in tests, as they can contain specific logic
@Wozniaxos @mikoscz thoughts?

Reply:
Yeah, you right, during some additional operation tests will be less efficient and this is also coupled with module.